test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,180 @@ def test_native_functions_always_exeutable(cql):
                     assert list(user_session.execute(f"SELECT max(a) FROM {table}")) == [(84,)]
                     assert list(user_session.execute(f"SELECT min(a) FROM {table}")) == [(3,)]
                     assert list(user_session.execute(f"SELECT sum(a) FROM {table}")) == [(102,)]
+
+@contextmanager
+def eventually_new_named_table(cql, table, schema, extra=""):
+    eventually_authorized(lambda: cql.execute(f'CREATE TABLE {table} ({schema}) {extra}'))
+    try:
+        yield table
+    finally:
+        eventually_authorized(lambda: cql.execute(f'DROP TABLE {table}'))
+
+# When a role has permissions to create a table, the role is given full
+# access to the table it just created (we call this feature "auto-grant").
+# The following tests check the auto-grant feature. The first test checks
+# that the base table created by CREATE TABLE is usable by the role creating
+# it. The second test is about the view created by CREATE MATERIALIZED VIEW
+# being accessible. The third test is about newly created CDC being accessible.
+def test_auto_grant_base(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key"
+            table = f'{test_keyspace}.{unique_name()}'
+            with eventually_new_named_table(user_session, table, schema):
+                # Check that the username was auto-granted SELECT permissions
+                eventually_authorized(lambda: user_session.execute(f'SELECT * FROM {table}'))
```

Comment:
why introduce "eventually" semantics to the test? can't we just do group0 read barrier (i.e. inside grant helper func) and then checks should work intermediately

Reply:
This is the same as I did in the other tests. I'm not sure if the "read barrier" API existed when I wrote this, but in any case, I couldn't get a clear answer from anyone whether it's enough, or not. The "group 0 read barrier" is only one of the problems that case cause us to get stale permissions - there is also the caching. The "eventually" stuff ensures that the test works in any case.

Reply:
ok, at some point we will want to make those checks stronger

Reply:
In my view, as soon as a "GRANT ..." returns, all later requests should see it. GRANT should have waited for "read barrier" or cache expiration (or somehow cause eviction of the the specific cache entry as part of the raft operation).
But we don't have this today, and the test needs to work today.